pkg/k8s/synced/resources.go

```diff
@@ -21,6 +24,25 @@ type Resources struct {
 	resources map[string]<-chan struct{}
 	// stopWait contains the result of cache.WaitForCacheSync
 	stopWait map[string]bool
+
+	// timeSinceLastEvent contains the time each resource last received an event.
+	timeSinceLastEvent map[string]time.Time
```

Comment:
Just wondering out loud here: how much memory is being dedicated to these maps? From what I can tell, these maps inside `Resources` live forever inside the `K8sWatcher` which is tied into the `Daemon` object. I presume we don't need their contents after we've synced up with K8s. 

If the memory is high enough, it could be a good cleanup to somehow make this `Resources` object tied to only the init phase of the `K8sWatcher` and then we can throw away the reference.

Anyway, very low urgency item.

Reply:
Yeah, good call.
I was thinking about just setting the map to be nil after the Sync step completes to let it be GC'd and avoiding setting event times when not necessary.

During init, the memory usage will be basically be k8s_resource_watchers x 64bits as we're just storing one time object per resource watcher so that'd be around 100bytes.